yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -126,15 +126,37 @@ export class Sequencer {
         this.log(`Failed to publish block`);
       }
 
-      // Publishes new unverified data for private txs to the network and awaits the tx to be mined
+      // Publishes new unverified data & contract data for private txs to the network and awaits the tx to be mined
       this.state = SequencerState.PUBLISHING_UNVERIFIED_DATA;
       const unverifiedData = UnverifiedData.join(validTxs.filter(isPrivateTx).map(tx => tx.unverifiedData));
+      const newContractData: CompleteContractData[] = validTxs
+        .filter(isPrivateTx)
+        .map(tx => {
+          // Currently can only have 1 new contract per tx
+          const newContract = tx.data?.end.newContracts[0];
+          if (newContract && tx.newContractPublicFunctions?.length) {
+            return new ContractData(
+              newContract.contractAddress,
+              newContract.portalContractAddress,
+              tx.newContractPublicFunctions,
+            );
+          }
+        })
+        .filter(cd => cd !== undefined) as CompleteContractData[]; // using `as` here since typescript doesn't automatically get .filter
```

Comment:
This cast can have unintended consequences later if we change what's in `CompleteContractData`. I'd suggest using [lodash.compact](https://lodash.com/docs/4.17.15#compact) instead, or something like [this type guard](https://stackoverflow.com/a/73605564).